source/shared/cpp/ObjectModel/MarkDownBlockParser.cpp

```diff
@@ -451,45 +473,127 @@ void LinkParser::CaptureLinkToken()
     m_parsedResult.AppendToTokens(codeGen);
 }
 
+
+// list items have form of ^-\s+ or \r-\s+
+bool ListParser::MatchNewListItem(std::stringstream &stream)
+{
+    if (IsHyphen(stream.peek()))
+    {
+        stream.get();
+        if (stream.peek() == ' ')
+        {
+            stream.unget();
+            return true;
+        }
+        stream.unget();
+    }
+    return false;
+}
+
+// if lines are seperated by more than two new lines,
+// they are new block item
+bool ListParser::MatchNewBlock(std::stringstream &stream)
+{
+    if (IsNewLine(stream.peek()))
+    {
+        do
+        {
+            stream.get();
+        } while (IsNewLine(stream.peek()));
+
+        return true;
+    }
+
+    return false;
+}
+
+bool ListParser::MatchNewOrderedListItem(std::stringstream &stream, std::string &number_string)
+{
+    do
+    {
+        number_string += stream.get();
+    } while (isdigit(stream.peek()));
+
+    if (stream.peek() == '.')
+    {
+        // ordered list syntax check complete
+        stream.unget();
+        return true;
+    }
+
+    return false;
+}
+// parse blocks that wasn't captured
+// if what we encounter is one of following items, start of new list, list item, or new block element,
+// we do not include in the current block, we return, and have it handled by the caller
+void ListParser::ParseSubBlocks(std::stringstream &stream)
+{
+    while (!stream.eof()) 
+    { 
+        if (IsNewLine(stream.peek()))
+        {
+            int newLineChar = stream.get();
+            // check if it is the start of new block items
+            if (isdigit(stream.peek()))
+            {
+                std::string number_string = "";
+                if (MatchNewOrderedListItem(stream, number_string))
+                {
+                    break;
+                }
+                else
+                { 
+                    m_parsedResult.AddNewTokenToParsedResult(number_string);
+                }
+            }
+            else if (MatchNewListItem(stream) || MatchNewBlock(stream))
+            {
+                break;
+            }
+
+            m_parsedResult.AddNewTokenToParsedResult(newLineChar);
+        }
+        ParseBlock(stream);
+    }
+}
 // list has form of -\s+ markdown block and a line break
 // this method checks such syntax
 void ListParser::Match(std::stringstream &stream) 
 { 
-    if (stream.peek() == '-')
+    // check for the mendatory space
```

Comment:
>mendatory [](start = 21, length = 9)

[nit] mandatory

Reply:
thank you

---
In reply to: [158142963](https://github.com/Microsoft/AdaptiveCards/pull/1053#discussion_r158142963) [](ancestors = 158142963)